bin/add_vcf_header_info_records.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+
+"""
+Add INFO fields to VCF Header
+"""
+
+import argparse
+import os
+from typing import List, Dict, Optional
+
+VALID_NUMBER_SPECIAL_CHARS = ("A", "R", "G", ".")
+VALID_INFO_TYPES = ("Integer", "Float", "Flag", "Character", "String")
+
+
+def main() -> None:
+    """
+    Add new INFO rows to VCF, output to new VCF file.
+    """
+    args = parse_arguments()
+
+    if not os.path.exists(args.vcf):
+        raise FileNotFoundError(f"Input VCF file '{args.vcf}' does not exist.")
+
+    processed_header_data = process_info_args(args.info)
+    info_lines_to_be_added = generate_info_lines(processed_header_data)
+    add_info_to_vcf(args.vcf, args.output, info_lines_to_be_added)
+
+
+def parse_arguments() -> argparse.Namespace:
+    """
+    Parse command-line arguments.
+
+    Returns:
+        argparse.Namespace: Parsed arguments.
+    """
+    parser = argparse.ArgumentParser(description="Add INFO fields to a VCF header.")
+
+    parser.add_argument(
+        "--vcf",
+        required=True,
+        help="Path to the input VCF file.",
+    )
+
+    parser.add_argument(
+        "--info",
+        action="append",
+        nargs=6,  # Expect exactly six arguments
+        metavar=("ID", "Number", "Type", "Description", "Source", "Version"),
+        required=True,
+        help=(
+            "Define an INFO field in the header. Provide exactly six arguments: "
+            "ID, Number, Type, Description, Source, and Version. Use an empty string "
+            "(e.g., '') for Source and Version if not applicable. This argument can "
+            "be repeated to add multiple INFO fields."
+        ),
+    )
+    parser.add_argument(
+        "--output",
+        required=True,
+        help="Path to the output VCF file.",
+    )
+
+    return parser.parse_args()
+
+
+def process_info_args(info_args: List[List[str]]) -> List[Dict[str, Optional[str]]]:
+    """
+    Process INFO arguments, ensuring valid Source and Version usage.
+
+    Args:
+        info_args (List[List[str]]): Parsed INFO arguments.
+
+    Returns:
+        List[Dict]: INFO fields as dictionaries.
+    """
+
+    id_: str
```

Comment:
Hmm. Here I think it would have been cleaner to use a simple class containing this set of data.

Then the function definition would be:

```
def process_info_args(info_args: List[InfoField]) -> List[Dict[str, Optional[str]]]:
```

And you could skip defining all types here

Anyway, posting more as a general thought. I am fine with leaving as is here

Reply:
i'd like to take it one step further and have something like a python3 container used for these kind of scripts with dependencies that do away with having to add new classes for storing VCF data or just opening VCF files for each util script.

basically a python container with a python version of the perl CMD::VCF libs.

Reply:
Yes, sure, I agree!